winbuild/build_prepare.py

```diff
@@ -569,7 +569,11 @@ def build_pillow():
         *prefs["header"],
         cmd_set("DISTUTILS_USE_SDK", "1"),  # use same compiler to build Pillow
         cmd_set("py_vcruntime_redist", "true"),  # always use /MD, never /MT
-        r'"{python_dir}\{python_exe}" setup.py build_ext --vendor-raqm --vendor-fribidi %*',  # noqa: E501
+        r'"{python_dir}\{python_exe}" -m pip install --upgrade pip',
```

Comment:
Is it really necessary to upgrade pip here?
I imagine it would be preferable to upgrade pip before calling this script instead (and recommend it in the README) to (slightly) speed up local development.

Reply:
Without any upgrade it was failing (for AppVeyor only) like:

<details>
<summary>Details</summary>



```

C:\pillow>"C:/Python38-x64\python.exe" -m pip install . --global-option="--vendor-raqm" --global-option="--vendor-fribidi" --global-option="install" 
Processing c:\pillow
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
c:\pillow\winbuild\build\build_pillow.cmd : C:\Python38-x64\lib\site-packages\pip\_internal\commands\install.py:283: UserWarning: Disabling all use of wheels due to the use of --build-options / --global-options / 
--install-options.
At line:1 char:1
+ c:\pillow\winbuild\build\build_pillow.cmd install
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (C:\Python38-x64...nstall-options.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
  cmdoptions.check_install_build_global(options)
  ERROR: Command errored out with exit status 2:
   command: 'C:\Python38-x64\python.exe' 'C:\Python38-x64\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\appveyor\AppData\Local\Temp\1\pip-build-env-cgb3zz5u\overlay' --no-warn-script-location 
--no-binary :all: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=67.8'
       cwd: None
  Complete output (44 lines):
  Collecting setuptools>=67.8
    Using cached https://files.pythonhosted.org/packages/03/20/630783571e76e5fa5f3e9f[2939](https://ci.appveyor.com/project/hugovk/Pillow/build/job/y6vupdxep0yqi8v5#L2939)8ca3ace377207b8196b54e0ffdf09f12c1/setuptools-67.8.0.tar.gz
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
  ERROR: Exception:
  Traceback (most recent call last):
    File "C:\Python38-x64\lib\site-packages\pip\_internal\cli\base_command.py", line 153, in _main
      status = self.run(options, args)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\commands\install.py", line 382, in run
      resolver.resolve(requirement_set)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\legacy_resolve.py", line 201, in resolve
      self._resolve_one(requirement_set, req)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\legacy_resolve.py", line 365, in _resolve_one
      abstract_dist = self._get_abstract_dist_for(req_to_install)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\legacy_resolve.py", line 312, in _get_abstract_dist_for
      abstract_dist = self.preparer.prepare_linked_requirement(
    File "C:\Python38-x64\lib\site-packages\pip\_internal\operations\prepare.py", line 223, in prepare_linked_requirement
      abstract_dist = _get_prepared_distribution(
    File "C:\Python38-x64\lib\site-packages\pip\_internal\operations\prepare.py", line 49, in _get_prepared_distribution
      abstract_dist.prepare_distribution_metadata(finder, build_isolation)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\distributions\source\legacy.py", line 37, in prepare_distribution_metadata
      self._setup_isolation(finder)
    File "C:\Python38-x64\lib\site-packages\pip\_internal\distributions\source\legacy.py", line 90, in _setup_isolation
      reqs = backend.get_requires_for_build_wheel()
    File "C:\Python38-x64\lib\site-packages\pip\_vendor\pep517\wrappers.py", line 151, in get_requires_for_build_wheel
      return self._call_hook('get_requires_for_build_wheel', {
    File "C:\Python38-x64\lib\site-packages\pip\_vendor\pep517\wrappers.py", line 255, in _call_hook
      raise BackendUnavailable(data.get('traceback', ''))
  pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
    File "C:\Python38-x64\lib\site-packages\pip\_vendor\pep517\_in_process.py", line 63, in _build_backend
      obj = import_module(mod_path)
    File "C:\Python38-x64\lib\importlib\__init__.py", line 127, in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
    File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
    File "<frozen importlib._bootstrap>", line 991, in _find_and_load
    File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
    File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
    File "<frozen importlib._bootstrap>", line 991, in _find_and_load
    File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
  ModuleNotFoundError: No module named 'setuptools'
  
  WARNING: You are using pip version 19.3.1; however, version 23.1.2 is available.
  You should consider upgrading via the 'python -m pip install --upgrade pip' command.
  ----------------------------------------
ERROR: Command errored out with exit status 2: 'C:\Python38-x64\python.exe' 'C:\Python38-x64\lib\site-packages\pip' install --ignore-installed --no-user --prefix 
'C:\Users\appveyor\AppData\Local\Temp\1\pip-build-env-cgb3zz5u\overlay' --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=67.8' Check the logs for full command output.
WARNING: You are using pip version 19.3.1; however, version 23.1.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
cd c:\pillow
%PYTHON%\%EXECUTABLE% selftest.py --installed
Traceback (most recent call last):
  File "selftest.py", line 6, in <module>
    from PIL import Image, features
ModuleNotFoundError: No module named 'PIL'
Command exited with code 1
```


</details>

https://ci.appveyor.com/project/hugovk/Pillow/build/job/y6vupdxep0yqi8v5

But we can move it `.appveyor.yml` instead. Updated!